### PR TITLE
Corrections for pair programming

### PR DESCRIPTION
**Pull request by kiragrammel**
* Provide current-user application-wide
* Synchronized Editor: Store all events and allow multiple tabs
* Simplify comparison of delta objects
* Set lastDeltaObject per file
* Add third question in pp exercise feedback
* Correct connection status if user used multiple windows

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -992,6 +992,14 @@ en:
...
     estimated_time_more_30: "more than 30 minutes"
     working_time: "Estimated time working on this exercise:"
+    reason_no_partner: I have not found a team partner.
+    reason_to_difficult_to_find_partner: It was too difficult to find a team partner.
+    reason_faster_alone: It was faster to work alone.
+    reason_not_working_with_strangers: I do not want to work with <i>strangers</i>.
+    reason_want_to_work_alone: I prefer to work alone.
+    reason_accidentally_alone: I accidentally worked alone on this exercise.
+    reason_other: <i>Other</i>
+    reason_work_alone: Why did you decide to solve the task alone?
```
Comment (MrSerth) [inline config/locales/en.yml:1002]:
```suggestion
    reason_work_alone: Why did you decide to solve this exercise alone?
```

File: app/assets/javascripts/editor/editor.js.erb [hunk 2]
```diff
@@ -335,11 +336,11 @@ var CodeOceanEditor = {
 
             // listener for autosave
             session.on("change", function (deltaObject, session) {
-                if (this.compareDeltaObjects(deltaObject, CodeOceanEditor.lastDeltaObject)) {
-                    CodeOceanEditor.lastDeltaObject = null;
+                if (this.compareDeltaObjects(deltaObject, this.lastDeltaObject_for_file[this.active_file.filename])) {
```
Comment (MrSerth) [inline app/assets/javascripts/editor/editor.js.erb:339]:
```suggestion
                if (this.compareDeltaObjects(deltaObject, this.lastDeltaObject_for_file.get(this.active_file.filename))) {
```

File: app/models/event/synchronized_editor.rb [hunk 3]
```diff
@@ -66,9 +64,19 @@ def self.create_for_editor_change(event, user, programming_group)
...
+      study_group_id: user.current_study_group_id,
+      action: message[:action],
+      status: message[:status]
+    )
+  end
+
   def self.data_attribute(event, delta)
     event[:delta] = {data: delta} if delta.present?
-    event.presence if event.present? # TODO: As of now, we are storing the `current_user_id` most of the times. Intended?
+    event.presence if event.present? # TODO: As of now, we are storing the `session_id` most of the times. Intended?
```
Comment (kiragrammel) [inline app/models/event/synchronized_editor.rb:79]: Should we remove the session id? I probably don't need it. Could only be used to check if multiple windows where used from a user.

Comment (MrSerth) [reply app/models/event/synchronized_editor.rb:81]: I'd go with it, and hence added df3ca1e4d802c12b2eea8135c570e1fe9629376b. Please give it a quick try; afterwards I'd proceed with merging this one.

Comment (kiragrammel) [reply app/models/event/synchronized_editor.rb:81]: Looks good! 👍🏻

**Pull request merged**